ISISComputingGroup/EPICS-LSICorrelator
# Refactor LSICorrelator as part of handover

## PR Justification

Changes made as part of [#6871](https://github.com/ISISComputingGroup/IBEX/issues/6871) within this PR aim to take steps to refactor the LSI Correlator to decouple from PCASPY while allowing me to become more familiar with the LSI Correlator as part of the handover process which took place. 

## Work Completed
- Remove unused methods and imports.
- Place static variables inside config file to clean up scripts.
- Create a basic UML class diagram to describe existing structure of LSI correlator.
- Add more descriptive docstrings to better document codebase.
- Update README.md to include relevant and useful recourses.

## Acceptance Criteria
- [ ] All tests in `master\tests.py` pass.
- [ ] All IOC system tests pass in `EPICS-IOC_Test_Framework\tests\lsicorr.py`.
- [ ] Docstrings are well formatted and contain no spelling mistakes.
- [ ] README.md is well structured and grammatically correct.
- [ ] Static variables placed in config are well structured.
- [ ] Changes in `data_file_interaction.py` are clear and concise.

### How to Test
* Checkout the `Ticket6893_refactoring` branch of LSI Correlator Support repo in `C:\Instrument\Apps\EPICS\support\lsicorr\master` and pull recent changes down
* Checkout the `master` branch of the LSI Correlcator IOC repo in `C:\Instrument\Apps\EPICS\support\lsicorr_vendor\master` making sure most recent changes have been pulled down.
* run `master\tests.py` from the LSI Correlator Support repo and ensure all tests pass.
* Run IOC system tests in `EPICS-IOC_Test_Framework\tests\lsicorr.py`, ensuring all tests pass.
* Continue with the rest of the review for LSI Correlator Support repo
